### PR TITLE
refactor(data-table-manager): restructure files and rewrite tests

### PR DESCRIPTION
**Pull request by emmenko**
This PR aims to refactor a bit the `<DataTableManager>` package to simplify the implementation a bit, as well as to rewrite the tests to have higher level integration tests.

This is a follow up of the initial implementation PR #1416

### CODE REVIEW

File: packages/components/data-table-manager/src/column-settings-manager/column-settings-manager.js [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+DropdownIndicator.displayName = 'DropdownIndicator';
+
+const Nothing = () => null;
+const selectInputComponents = {
+  Option: Nothing,
+  Menu: Nothing,
+  DropdownIndicator,
+};
+
+export const ColumnSettingsManager = (props) => {
```
Comment (emmenko) [inline packages/components/data-table-manager/src/column-settings-manager/column-settings-manager.js:89]: Renaming this to `ColumnSettingsManager`

File: packages/components/data-table-manager/src/column-settings-manager/column-settings-manager.js [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+  );
+
+  return (
+    <SettingsContainer
+      title={messages.title}
+      closeButtonLabel={messages.closeButtonLabel}
+      onClose={props.onClose}
+      primaryButton={props.primaryButton}
+      secondaryButton={props.secondaryButton}
+    >
```
Comment (emmenko) [inline packages/components/data-table-manager/src/column-settings-manager/column-settings-manager.js:134]: Extracted common UI parts into a shared component

File: packages/components/data-table-manager/src/data-table-settings/data-table-settings.js [hunk 3]
```diff
@@ -102,7 +105,7 @@ const DataTableSettings = (props) => {
         )}
       </Spacings.Inline>
       {openedPanelId === DISPLAY_SETTINGS && (
-        <DensityManager
+        <DisplaySettingsManager
```
Comment (emmenko) [inline packages/components/data-table-manager/src/data-table-settings/data-table-settings.js:108]: Renamed this as well, to have consistent naming

File: packages/components/data-table-manager/src/display-settings-manager/display-settings-manager.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    : DENSITY_DEFAULT;
+
+  return (
+    <SettingsContainer
+      title={messages.title}
+      closeButtonLabel={messages.closeButtonLabel}
+      onClose={props.onClose}
+      primaryButton={props.primaryButton}
+      secondaryButton={props.secondaryButton}
+    >
```
Comment (emmenko) [inline packages/components/data-table-manager/src/display-settings-manager/display-settings-manager.js:34]: The shared component

File: packages/components/data-table-manager/src/data-table-manager.spec.js [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+   */
+  onSettingsChange: PropTypes.func,
+  /**
+   * A React node for rendering additional information within the table manager.
+   */
+  topBar: PropTypes.node,
+};
+DataTableManager.displayName = 'DataTableManager';
+
+export default DataTableManager;
```
Comment (tdeekens) [inline packages/components/data-table-manager/src/data-table-manager.spec.js:241]: Why is this file called `.spec`?

Comment (emmenko) [reply packages/components/data-table-manager/src/data-table-manager.spec.js:241]: Don't bother reviewing this for now

File: packages/components/data-table-manager/src/data-table-manager.spec.js [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+    const selectDropdown = await screen.findByLabelText(
+      'Open table manager dropdown'
+    );
+
+    fireEvent.focus(selectDropdown);
+    fireEvent.change(selectDropdown, { target: { value: 'column' } });
+    fireEvent.keyDown(selectDropdown, { key: 'Enter', keyCode: 13, which: 13 });
+
+    await screen.findByText('Column Manager');
+  });
```
Comment (emmenko) [inline packages/components/data-table-manager/src/data-table-manager.spec.js:92]: Alright, this is how I would like to write the tests.

I just added a bunch of tests to check for the initial rendering. Interaction tests will follow (possibly also in a separate PR to keep things smaller).

File: packages/components/data-table-manager/src/data-table-manager.spec.js [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+    );
+
+    fireEvent.focus(selectDropdown);
+    fireEvent.change(selectDropdown, { target: { value: 'display' } });
+    fireEvent.keyDown(selectDropdown, { key: 'Enter', keyCode: 13, which: 13 });
+
+    await screen.findByText('Table layout settings');
+
+    expect(
+      screen.queryByLabelText('Select radio option: display full text')
```
Comment (emmenko) [inline packages/components/data-table-manager/src/data-table-manager.spec.js:70]: To help with these selectors, I'm using the `AccessibleHidden` component

File: packages/components/data-table-manager/src/data-table-settings/data-table-settings.js [hunk 8]
```diff
@@ -89,8 +95,13 @@ const DataTableSettings = (props) => {
         <div>{props.topBar}</div>
         {dropdownOptions.length > 0 && (
           <SelectContainer>
+            <AccessibleHiden>
+              <label htmlFor="table-settings-dropdown">
+                {'Open table manager dropdown'}
```
Comment (tdeekens) [inline packages/components/data-table-manager/src/data-table-settings/data-table-settings.js:100]:
```suggestion
                Open table manager dropdown
```

Should work too I think.

File: packages/components/data-table-manager/src/display-settings-manager/display-settings-manager.js [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+      <Grid gridGap={vars.spacingM} gridTemplateColumns="repeat(2, 1fr)">
+        <Grid.Item>
+          <Spacings.Stack scale="s">
+            <FieldLabel
+              title={intl.formatMessage(messages.textWrappingSubtitle)}
+            />
+            <AccessibleHiden>
+              <label htmlFor="text-wrapping-0">
+                {'Select radio option: display full text'}
```
Comment (tdeekens) [inline packages/components/data-table-manager/src/display-settings-manager/display-settings-manager.js:44]:
```suggestion
                Select radio option: display full text
```

File: packages/components/data-table-manager/src/display-settings-manager/display-settings-manager.js [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+              </label>
+              <label htmlFor="text-wrapping-1">
+                {'Select radio option: display full previews'}
```
Comment (tdeekens) [inline packages/components/data-table-manager/src/display-settings-manager/display-settings-manager.js:47]:
```suggestion
                Select radio option: display full previews
```

File: packages/components/data-table-manager/src/display-settings-manager/display-settings-manager.js [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+          </Spacings.Stack>
+        </Grid.Item>
+        <Grid.Item>
+          <Spacings.Stack scale="s">
+            <FieldLabel
+              title={intl.formatMessage(messages.densityDisplaySubtitle)}
+            />
+            <AccessibleHiden>
+              <label htmlFor="density-display-0">
+                {'Select radio option: density default'}
```
Comment (tdeekens) [inline packages/components/data-table-manager/src/display-settings-manager/display-settings-manager.js:74]:
```suggestion
                Select radio option: density default
```

File: packages/components/data-table-manager/src/display-settings-manager/display-settings-manager.js [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+              </label>
+              <label htmlFor="density-display-1">
+                {'Select radio option: density compact'}
```
Comment (tdeekens) [inline packages/components/data-table-manager/src/display-settings-manager/display-settings-manager.js:77]:
```suggestion
                Select radio option: density compact
```

**Review by tdeekens — APPROVED** [review]

File: packages/components/data-table-manager/src/data-table-settings/data-table-settings.js [hunk 13]
```diff
@@ -3,6 +3,7 @@ import React, { useState } from 'react';
 import invariant from 'tiny-invariant';
 import { useIntl } from 'react-intl';
 import styled from '@emotion/styled';
+import AccessibleHiden from '@commercetools-uikit/accessible-hidden';
```
Comment (jonnybel) [inline packages/components/data-table-manager/src/data-table-settings/data-table-settings.js:6]:
```suggestion
import AccessibleHidden from '@commercetools-uikit/accessible-hidden';
```

This confused me when using CTRL+F 😛

**Pull request merged**